### ISSUE TITLE
Not filled req->crlf

### ISSUE THREAD

**Issue by vladtcvs** [natsys/tempesta#326]
At the begining of `frang_http_req_process()` add:
```
TFW_DBG("%s: =========> crlf->ptr = %p, crlf->len = %d, crlf->flags = %x\n",
        __func__, req->crlf.ptr, req->crlf.len, req->crlf.flags);
```

On the host that runs Tempesta do:
```
telnet 127.0.0.1 80<enter>
GET http://localhost/ HTTP/1.1<enter>
Host: localhost<enter>
<enter>
```

The result is:
```
[tempesta]   frang_http_req_process: =========> crlf->ptr =           (null), crlf->len = 0, crlf->flags = 0
[tempesta]   frang_http_req_process: =========> crlf->ptr =           (null), crlf->len = 0, crlf->flags = 0
[tempesta]   frang_http_req_process: =========> crlf->ptr = ffff8801172d48d0, crlf->len = 0, crlf->flags = 0
```

So, crlf->ptr gets assigned to, but length remains zero.
Below is the full trace of the request/response.

```
[sync_sockets]   ss_tcp_data_ready: sk ffff880116d7f4c0, sk->sk_socket           (null), state (Established)
[tempesta]   Link new msg ffff880105b9f020 with connection ffff880105bf3a70
[tempesta]   Add skb ffff880116faae88 to message ffff880105b9f020
[tempesta]   GFSM exec fsm 0, state 0x2000
[tempesta]     Received 32 client data bytes on conn=ffff880105bf3a70 msg=ffff880105b9f020
[tempesta]   parse 32 client data bytes (GET http://localhost/ HTTP/1.1
) on req=ffff880105b9f020
[tempesta]     Request parsed: len=32 parsed=32 msg_len=32 res=-1
[tempesta]   GFSM move from 0x2000 to 0x1
[tempesta]   GFSM switch from fsm 0 at state 1 to fsm 2 at state 0x400
[tempesta]   GFSM exec fsm 2, state 0x2400
[tempesta]   frang_http_req_process: =========> crlf->ptr =           (null), crlf->len = 0, crlf->flags = 0
[sync_sockets]   ss_tcp_data_ready: sk ffff880116d7f4c0, sk->sk_socket           (null), state (Established)
[tempesta]   Add skb ffff880116faac48 to message ffff880105b9f020
[tempesta]   GFSM exec fsm 0, state 0x2001
[tempesta]     Received 17 client data bytes on conn=ffff880105bf3a70 msg=ffff880105b9f020
[tempesta]   parse 17 client data bytes (Host: localhost
) on req=ffff880105b9f020
[tempesta]     Request parsed: len=17 parsed=17 msg_len=49 res=-1
[tempesta]   GFSM move from 0x2001 to 0x1
[tempesta]   GFSM switch from fsm 0 at state 1 to fsm 2 at state 0x400
[tempesta]   GFSM exec fsm 2, state 0x2400
[tempesta]   frang_http_req_process: =========> crlf->ptr =           (null), crlf->len = 0, crlf->flags = 0
[sync_sockets]   ss_tcp_data_ready: sk ffff880116d7f4c0, sk->sk_socket           (null), state (Established)
[tempesta]   Add skb ffff880116faa348 to message ffff880105b9f020
[tempesta]   GFSM exec fsm 0, state 0x2001
[tempesta]     Received 2 client data bytes on conn=ffff880105bf3a70 msg=ffff880105b9f020
[tempesta]   parse 2 client data bytes (
) on req=ffff880105b9f020
[tempesta]     Request parsed: len=2 parsed=2 msg_len=51 res=0
[tempesta]   GFSM move from 0x2001 to 0x2
[tempesta]   GFSM switch from fsm 0 at state 2 to fsm 2 at state 0x400
[tempesta]   GFSM exec fsm 2, state 0x2400
[tempesta]   frang_http_req_process: =========> crlf->ptr = ffff8801172d48d0, crlf->len = 0, crlf->flags = 0
[tempesta]   Matching request: ffff880105b9f020, list: ffff8801038c3020
[tempesta]     rule: ffff8801038c3058, field: 0x1, op: 0x1, arg:1:1'*'
[tempesta]   tfw_sched_http: use server group: 'default'
[sync_sockets]   skb fragmentation (len=28 pspt=ffff8801172d48d0, skb: head=ffff8801172d4800 data=ffff8801172d48d0 tail=ffff8801172d48d2 end=ffff8801172d4ec0 len=2 data_len=0 truesize=2368 nr_frags=0 frag_list=          (null))...
[sync_sockets]   __skb_fragment: res=ffff880103e69000, skb: head=ffff8801172d4800 data=ffff8801172d48d0 tail=ffff8801172d48d0 end=ffff8801172d4ec0 len=30 data_len=30 truesize=2398 nr_frags=2 frag_list=          (null))
[tempesta]     added X-Forwarded-For header for 127.0.0.1
[sync_sockets]   skb fragmentation (len=24 pspt=ffff8801172d48d0, skb: head=ffff8801172d4800 data=ffff8801172d48d0 tail=ffff8801172d48d0 end=ffff8801172d4ec0 len=30 data_len=30 truesize=2398 nr_frags=2 frag_list=          (null))...
[sync_sockets]   __skb_fragment: res=ffff880103e6901c, skb: head=ffff8801172d4800 data=ffff8801172d48d0 tail=ffff8801172d48d0 end=ffff8801172d4ec0 len=54 data_len=54 truesize=2422 nr_frags=2 frag_list=          (null))
[sync_sockets]   ss_send: sk ffff88010600ca80, sk->sk_socket           (null), state (Established)
[sync_sockets]   ss_send: entail skb=ffff880100d595c0 data_len=0 len=32
[sync_sockets]   ss_send: entail skb=ffff880100d59700 data_len=0 len=17
[sync_sockets]   ss_send: entail skb=ffff880100d59480 data_len=54 len=54
[sync_sockets]   ss_send: sk=ffff88010600ca80 is_queue_empty=0 tcp_send_head(sk)=ffff880100d595c0 sk->sk_state=1
[sync_sockets]   ss_tcp_data_ready: sk ffff88010600ca80, sk->sk_socket           (null), state (Established)
[tempesta]   Link new msg ffff88011713a020 with connection ffff880105afcc00
[tempesta]   Add skb ffff880100d59200 to message ffff88011713a020
[tempesta]   GFSM exec fsm 0, state 0x2000
[tempesta]     received 858 server data bytes on conn=ffff880105afcc00 msg=ffff88011713a020
[tempesta]   parse 62 server data bytes (HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Date: Thu, 22 O) on resp=ffff88011713a020
[tempesta]   parse 796 server data bytes (ct 2015 16:17:04 GMT
Content-Type: text/html
Content-Length: 612
Last-Modified: Tue, 04 Mar 2014 11:46:45 GMT
Connection: keep-alive
ETag: "5315bd25-264"
Accept-Ranges: bytes

<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>
) on resp=ffff88011713a020
[tempesta]     response parsed: len=858 parsed=858 res=0
[tempesta]   GFSM move from 0x2000 to 0x4
[tempesta]   GFSM move from 0x2004 to 0x5
[sync_sockets]   ss_send: sk ffff880116d7f4c0, sk->sk_socket           (null), state (Established)
[sync_sockets]   ss_send: entail skb=ffff880100d59480 data_len=796 len=858
[sync_sockets]   ss_send: sk=ffff880116d7f4c0 is_queue_empty=0 tcp_send_head(sk)=ffff880100d59480 sk->sk_state=1
```

**Comment by krizhanovsky** [comment]
CRLF must be 2 bytes in length, so probably we just should initialize the TfwStr.len by 2

**Comment by vladtcvs** [comment]
Fixed in #329

**Comment by vladtcvs** [comment]
fix merged
